### PR TITLE
Added fetch_start functionality

### LINKED ISSUE #21258: PerformanceResourceTiming: FetchStart
In components/net/http_loader.rs::http_fetch we need to set fetch_start.

On getting, the fetchStart attribute MUST return as follows:

1. The time immediately before the user agent starts to fetch the final resource in the redirection, if there are HTTP redirects or equivalent.
2. The time immediately before the user agent starts to fetch the resource otherwise.

https://w3c.github.io/resource-timing/#dom-performanceresourcetiming-fetchstart

### PR DESCRIPTION
**Pull request by aditj**
Added the fetch_start functionality in http_loader.rs (http_fetch function)

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #21258 (GitHub issue number if applicable)

- [X] There are tests for these changes OR
- [ ] These changes do not require tests because ___

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/22714)

### CODE REVIEW

**Comment by SimonSapin** [comment]
The `.wpt-update.lock` file presumably shouldn’t be added. (Should we even add it to some `.gitignore` file?)

r? @avadacatavra

File: components/net/http_loader.rs [hunk 1]
```diff
@@ -500,6 +500,11 @@ pub fn http_fetch(
     *done_chan = None;
     // Step 1
     let mut response: Option<Response> = None;
+    context
+        .timing
+        .lock()
+        .unwrap()
+        .set_attribute(ResourceAttribute::FetchStart);
```
Comment (jdm) [inline components/net/http_loader.rs]: We should initialize our fetchStart value in https://github.com/servo/servo/blob/49cf3dd3d337689ac72bea1a8ffc5c3e89a2616c/components/net/fetch/methods.rs#L90 instead, so it encompasses all kinds of requests.

File: components/net/http_loader.rs [hunk 2]
```diff
@@ -568,6 +573,13 @@ pub fn http_fetch(
         // TODO(#21256) maybe set redirect_start if this resource initiates the redirect
         // TODO(#21254) also set startTime equal to either fetch_start or redirect_start
         //   (https://w3c.github.io/resource-timing/#dfn-starttime)
+        if request.redirect_count > 0 && request.headers.contains_key(header::LOCATION) {
```
Comment (jdm) [inline components/net/http_loader.rs]: We want to check the response headers, not the request ones. However, I think if we add this `set_attribute` call to the `http_redirect_fetch` function, that should automatically cover the cases we care about.

File: tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/update/.wpt-update.lock [hunk 3]
```diff
@@ -0,0 +1,366 @@
+ccopy_reg
```
Comment (jdm) [inline tests/wpt/web-platform-tests/tools/wptrunner/wptrunner/update/.wpt-update.lock]: Please remove this file and run `./mach update-manifest`.

**Review by jdm — CHANGES_REQUESTED** [review]
This looks like a good start!

File: components/net/http_loader.rs (hunk 2 above)
Comment (jdm) [reply components/net/http_loader.rs]: I do not think the header check is needed. If we unconditionally update fetchStart each time http_redirect_fetch is called, then the last updated value will be the correct one.

**Comment by jdm** [comment]
```
error[E0433]: failed to resolve: use of undeclared type or module `ResourceAttribute`
  --> components/net/fetch/methods.rs:96:24
   |
96 |         .set_attribute(ResourceAttribute::FetchStart);
   |                        ^^^^^^^^^^^^^^^^^ use of undeclared type or module `ResourceAttribute`
error: aborting due to previous error
```
Also you didn't run `./mach update-manifest` like I asked.

**Comment by jdm** [comment]
To be clear, the results of running `./mach update-manifest` were not included in the most recent commit.

File: components/net/fetch/methods.rs [hunk 4]
```diff
@@ -89,6 +89,11 @@ pub type DoneChannel = Option<(Sender<Data>, Receiver<Data>)>;
 /// [Fetch](https://fetch.spec.whatwg.org#concept-fetch)
 pub fn fetch(request: &mut Request, target: Target, context: &FetchContext) {
     fetch_with_cors_cache(request, &mut CorsCache::new(), target, context);
+    context
+        .timing
+        .lock()
+        .unwrap()
+        .set_attribute(ResourceAttribute::FetchStart);
```
Comment (jdm) [inline components/net/fetch/methods.rs]: This needs to happen before calling `fetch_with_cors_cache` or it will execute after the fetch is complete.

**Review by jdm — CHANGES_REQUESTED** [review]

**Comment by aditj** [comment]
> To be clear, the results of running `./mach update-manifest` were not included in the most recent commit.

I ran ./mach update-manifest but I dont think the .json file has changes w.r.t. the Servo Master branch , also should I update the test expectations?

File: components/net/fetch/methods.rs [hunk 5]
```diff
@@ -88,6 +89,11 @@ pub type DoneChannel = Option<(Sender<Data>, Receiver<Data>)>;
 
 /// [Fetch](https://fetch.spec.whatwg.org#concept-fetch)
 pub fn fetch(request: &mut Request, target: Target, context: &FetchContext) {
+    context
```
Comment (jdm) [inline components/net/fetch/methods.rs]: Let's add a comment here like `// Step 7 of https://w3c.github.io/resource-timing/#processing-model`

File: components/net/http_loader.rs [hunk 6]
```diff
@@ -694,6 +694,11 @@ pub fn http_redirect_fetch(
         },
         Some(Ok(url)) => url,
     };
+    context
```
Comment (jdm) [inline components/net/http_loader.rs]: Let's add a comment here like `// Step 1 of https://w3c.github.io/resource-timing/#dom-performanceresourcetiming-fetchstart`

**Review by jdm — CHANGES_REQUESTED** [review]

**Comment by aditj** [comment]
@ferjm Are the test expectations fine now?

**Comment by ferjm** [comment]
Let's see if they are.

@bors-servo try=wpt

**Comment by jdm** [comment]
@bors-servo retry
* #13480

**Pull request merged**